lib/services/episode.ts

```diff
@@ -0,0 +1,68 @@
+import type { Tables } from '@/types/supabase/database';
+
+import { z } from 'zod';
+
+import type { PodcastIndexEpisodeType } from './podcast-index/fetch-episodes';
+
+import { DatabaseError } from '../errors';
+import { createSupabaseServiceClient } from './supabase/service';
+
+const serializedEpisodesSchema = z.array(
+  z.object({
+    audio_url: z.string().min(1),
+    description: z.string().nullable(),
+    duration: z.number().positive(),
+    episode_number: z.number().nullable(),
+    image: z.string(),
+    podcast_index_guid: z.string().min(1),
+    published_date: z.coerce.date(),
+    published_date_pretty: z.string().min(1),
+    show: z.number().positive(),
+    title: z.string().min(1),
+  }),
+);
+
+const mapEpisodes = (episodes: PodcastIndexEpisodeType[], showId: number) => {
+  const serializedEpisodes = episodes.map((item) => {
+    return {
+      audio_url: item.enclosureUrl,
+      description: item.description,
+      duration: item.duration,
+      episode_number: item.episode,
+      image: item.feedImage || item.image,
+      podcast_index_guid: item.guid,
+      published_date: item.datePublished,
+      published_date_pretty: item.datePublishedPretty,
+      show: showId,
+      title: item.title,
+    };
+  });
+
+  const validation = serializedEpisodesSchema.safeParse(serializedEpisodes);
+
+  if (!validation.success) {
+    throw validation.error;
```

Comment:
suggestion: instead of throwing
- we can return `null` or something here
- then `filter` non-null episodes

or just trust `podcast index` and leave it as it is 🤔

Reply:
I think this much better option. As an addition, we can log the failing validations in the future to understand what is wrong and if that's something we can fix to import the episode.